zerver/lib/message.py

```diff
@@ -1430,3 +1431,38 @@ def set_visibility_policy_possible(user_profile: UserProfile, message: Message)
 def remove_single_newlines(content: str) -> str:
     content = content.strip("\n")
     return re.sub(r"(?<!\n)\n(?!\n|[-*] |[0-9]+\. )", " ", content)
+
+
+def event_recipient_ids_for_action_on_messages(
+    messages: list[Message], *, exclude_long_term_idle_users: bool = False
+) -> set[int]:
+    assert len(messages) > 0
+    message_ids = [message.id for message in messages]
+
+    def get_user_ids_having_usermessage_row_for_messages(message_ids: list[int]) -> set[int]:
+        return set(
+            UserMessage.objects.filter(message_id__in=message_ids).values_list(
+                "user_profile_id", flat=True
+            )
```

Comment:
This could be optimized with `exclude_long_term_idle_users` too, I think.

Reply:
Done